src/Streamly/Internal/FS/Event/Windows.hs

```diff
@@ -472,13 +461,16 @@ closePathHandleStream =
 --
 -- /Pre-release/
 --
-watchWith :: (Config -> Config) -> NonEmpty (Array Word8) -> Stream IO Event
+watchWith :: (Config -> Config) -> NonEmpty Path -> Stream IO Event
 watchWith f paths =
     S.bracketIO before after (S.parConcatMap (eager True) eventStreamAggr)
 
     where
 
-    before = return $ pathsToHandles (utf8ToStringList paths) $ f defaultConfig
+    before =
+        return
+            $ pathsToHandles (NonEmpty.map Path.toString paths)
```

Comment:
We should not use Strings in pathsToHandles. Should use the raw path directly instead. We should raise an issue to fix this.